lib/active_merchant/billing/gateways/decidir.rb

```diff
@@ -239,9 +240,11 @@ def message_from(success, response)
         return response['message'] if response['message']
 
         message = nil
-
         if error = response.dig('status_details', 'error')
           message = error.dig('reason', 'description')
+          if detail = error.dig('type')
+            message += ' | ' + detail
```

Comment:
Just wondering, could the code get to this point if `error.dig('reason', 'description')` returned `nil`? If so, this line would blow up.

Would it be worth doing something like `message = [message, detail].compact.join(' | ')` to guard against that?

Reply:
Since the `if` statement is within the scope of `if error = response.dig('status_details', 'error')`, it would not get to this point if that is  `nil`. I imagine there is a cleaner way to do this, but without knowing a lot of tricks in ruby, I just took the easy route 😄

Reply:
Ah no problem. I was saying if `message` was set to `nil` on line 244, which I don't think any `if` statements catch.

Another idea might be to just do `message = "#{message} | #{detail}", which'd handle if `message` was `nil` (I think that's kinda like f-strings in Python, if I remember that right 😄 )

Reply:
@naashton Still kinda wondering about this part. If the response had this kinda shape:
```
{"status_details"=>{"error"=>{"type"=>"typy"}}}
```
it'd end up raising an error on line 246, since `message` would be `nil` there.

What do you think about something like this (replacing lines 244-247)?
```
message = [
  error.dig('reason', 'description'),
  error['type']
].compact.join(' | ')
```
Just a thought. What think ye?

Reply:
Ah, I see what you are saying now. I'll make this change.

Reply:
Just pushed that change up. I'm using the fancy string format method: `message = "#{msg} | #{detail}"` 😄